### PR TITLE
CI: unpin pyarrow, fix failing test

### LINKED ISSUE #51146: CI: pyarrow 11 broke ci
The new pyarrow fails 4 tests. We already pinned to < 11 to keep ci green.

https://app.circleci.com/pipelines/github/pandas-dev/pandas/23874/workflows/5f47d077-f9c7-46d1-8fe2-0a6977ef5444/jobs/54659

### PR DESCRIPTION
**Pull request by lukemanley**
- [x] closes #51146
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Unpins pyarrow and fixes `test_from_sequence_of_strings_pa_array` which was xfailed for pyarrow duration types.

pyarrow 11.0 introduces support for casting from duration to string:

```
In [4]: pa.array([1], type=pa.duration("s")).cast(pa.string())
Out[4]: 
<pyarrow.lib.StringArray object at 0x7f77ed6cc9a0>
[
  "1"
]
```

pyarrow < 11.0 raises:
```
ArrowNotImplementedError: Unsupported cast from duration[s] to utf8 using function cast_string
```

### CODE REVIEW

File: pandas/core/arrays/arrow/array.py [hunk 1]
```diff
@@ -258,9 +258,19 @@ def _from_sequence_of_strings(
...
         elif pa.types.is_duration(pa_type):
-            from pandas.core.tools.timedeltas import to_timedelta
+            try:
+                # GH51175: test_from_sequence_of_strings_pa_array
+                # attempt to parse as int64 reflecting pyarrow's
+                # duration to string casting behavior
+                if isinstance(strings, (pa.Array, pa.ChunkedArray)):
+                    scalars = strings
+                else:
+                    scalars = pa.array(strings, type=pa.string(), from_pandas=True)
```
Comment (phofl) [inline pandas/core/arrays/arrow/array.py:268]: Can we cast this directly? This might make 2 copies?

Comment (lukemanley) [reply pandas/core/arrays/arrow/array.py:268]: Do you mean cast from string -> duration directly? If so, no, pyarrow 11 doesn't seem to allow that:

```
import pyarrow as pa

durations = pa.array([1], type=pa.duration("s"))
strings = durations.cast(pa.string())

strings.cast(pa.duration("s")) # -> raises ArrowNotImplementedError
```

Comment (phofl) [reply pandas/core/arrays/arrow/array.py:268]: No I meant:

```
scalars = pa.array(strings, type=pa.string(), from_pandas=True)
scalars = scalars.cast(pa.int64())
```

in the else case. Any reason we are not using int64 directly?

Comment (lukemanley) [reply pandas/core/arrays/arrow/array.py:268]: sorry, I'm confused with your comment. The `if` block needs to be converted from string -> int64 as well so i'm not following why you would put the cast to int64 inside the `else` block?

```
if isinstance(strings, (pa.Array, pa.ChunkedArray)):
    scalars = strings
else:
    scalars = pa.array(strings, type=pa.string(), from_pandas=True)
scalars = scalars.cast(pa.int64())
```

Comment (phofl) [reply pandas/core/arrays/arrow/array.py:268]: I am talking about doing

```
scalars = pa.array(strings, type=pa.int64(), from_pandas=True)
```
in the else block.
Sorry if that wasn't clear. Seems a bit confusing that we create the array with string before casting to int64

Comment (lukemanley) [reply pandas/core/arrays/arrow/array.py:268]: oh I see. pyarrow does not allow that:

```
pa.array(["1"], type=pa.int64(), from_pandas=True)

# ArrowInvalid: Could not convert '1' with type str: tried to convert to int64
```

Comment (phofl) [reply pandas/core/arrays/arrow/array.py:268]: Yikes, thx

Comment (mroeschke) [inline pandas/core/arrays/arrow/array.py:268]: IMO (slightly) I think we should still prefer to use `to_timedelta` first if a non-pyarrow `scalars` was passed over the `cast`

Comment (lukemanley) [reply pandas/core/arrays/arrow/array.py:268]: I think using `to_timedelta` first would be incorrect in some cases. For instance, if passed "1000 seconds" as
```
ArrowExtensionArray._from_sequence_of_strings(["1000"], dtype=pa.duration("s"))
```

`pd.to_timedelta(["1000"])` is interpreted as `1000 ns` (`to_timedelta` does not allow specifying `unit` with a string input and assumes nanoseconds).

Comment (lukemanley) [reply pandas/core/arrays/arrow/array.py:268]: in fact, the example provided above errors on main due to assuming ns units:

```
import pyarrow as pa

ArrowExtensionArray._from_sequence_of_strings(["1000"], dtype=pa.duration("s"))

# ArrowInvalid: Casting from duration[ns] to duration[s] would lose data: 1000
```

Comment (jbrockmendel) [reply pandas/core/arrays/arrow/array.py:268]: FWIW there's a corner case with iNaT that pyarrow will handle differently from to_timedelta

Comment (lukemanley) [reply pandas/core/arrays/arrow/array.py]: Good point re: `NaT`. I made some changes which now call `to_timedelta` first and only attempts the int conversion when the pyarrow unit is non-nano. It respects the null mask from `to_timedelta` so I think the `NaT` edge case is handled. I added a test for it. 

It all feels a bit messy, but I don't see another option ATM. Open to suggestions.

**Review by mroeschke — COMMENTED** [review]
Looks okay. Just a merge conflict

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @lukemanley

**Comment by phofl** [comment]
I guess we should add 11 to our build matrix?

**Comment by mroeschke** [comment]
Good point. Otherwise 11 is only running on the macos and windows builds. Want to open a PR?

**Pull request merged**